matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/typing/SendTypingTask.kt

```diff
@@ -38,12 +43,21 @@ internal interface SendTypingTask : Task<SendTypingTask.Params, Unit> {
 internal class DefaultSendTypingTask @Inject constructor(
         private val roomAPI: RoomAPI,
         @UserId private val userId: String,
-        private val globalErrorReceiver: GlobalErrorReceiver
+        private val globalErrorReceiver: GlobalErrorReceiver,
+        private val matrixConfiguration: MatrixConfiguration,
+        private val session: Session,
+        private val cryptoService: CryptoService
 ) : SendTypingTask {
 
     override suspend fun execute(params: SendTypingTask.Params) {
         delay(params.delay ?: -1)
 
+        if (params.isTyping
+                && session.getRoom(params.roomId)?.isEncrypted().orFalse()
+                && matrixConfiguration.outboundSessionKeySharingStrategy == OutboundSessionKeySharingStrategy.WhenTyping) {
```

Comment:
Same remark here. I find it a bit strange to wire this here in the task.

Reply:
Yeah, this is not the right place to do that

Reply:
More importantly, user can decide to disable sending typing event from the setting, it should not affect what is brought by this PR

Reply:
Done. Moved to RoomDetailViewModel, like element-ios implementation.